.github/workflows/discord-notification.yml

```
@@ -0,0 +1,107 @@
+name: Discord Notification
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, closed]
+  push:
+    branches: [main, develop] # Add your target branches
+  issues:
+    types: [opened, closed, reopened]
+  workflow_run:
+    workflows: ["CI", "Tests"] # Add your other workflow names
+    types: [completed]
+
+jobs:
+  notify:
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-latest
    concurrency:
        group: discord-notification-${{ github.ref }}
        cancel-in-progress: true
```

Add concurrency controls to prevent notification spam during rapid commits